teams_accounts.go

```diff
@@ -0,0 +1,131 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"net/http"
+	"time"
+)
+
+type TeamsAccount struct {
+	GatewayID    string `json:"gateway_tag"`   // Internal teams ID
+	ProviderName string `json:"provider_name"` // Auth provider
+	CloudflareID string `json:"id"`            // cloudflare account ID
+}
+
+// TeamsAccountResponse is the API response, containing information on teams account
+type TeamsAccountResponse struct {
+	Response
+	Result TeamsAccount `json:"result"`
+}
+
+// TeamsAccountResponse is the API response, containing information on teams account config
+type TeamsConfigResponse struct {
+	Response
+	Result TeamsConfiguration `json:"result"`
+}
+
+// TeamsConfiguration data model.
+type TeamsConfiguration struct {
+	Settings  TeamsAccountSettings `json:"settings"`
+	CreatedAt time.Time            `json:"created_at,omitempty"`
+	UpdatedAt time.Time            `json:"updated_at,omitempty"`
+}
+
+type TeamsAccountSettings struct {
+	Antivirus   *TeamsAntivirus   `json:"antivirus,omitempty"`
+	TLSDecrypt  *TeamsTLSDecrypt  `json:"tls_decrypt,omitempty"`
+	ActivityLog *TeamsActivityLog `json:"activity_log,omitempty"`
+	BlockPage   *TeamsBlockPage   `json:"block_page,omitempty"`
+}
+
+type TeamsAntivirus struct {
+	EnabledDownloadPhase bool `json:"enabled_download_phase,omitempty"`
+	EnabledUploadPhase   bool `json:"enabled_upload_phase,omitempty"`
+	FailClosed           bool `json:"fail_closed,omitempty"`
+}
+
+type TeamsTLSDecrypt struct {
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+type TeamsActivityLog struct {
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+type TeamsBlockPage struct {
+	Enabled         bool   `json:"enabled,omitempty"`
+	FooterText      string `json:"footer_text,omitempty"`
+	HeaderText      string `json:"header_text,omitempty"`
+	LogoPath        string `json:"logo_path,omitempty"`
+	BackgroundColor string `json:"background_color,omitempty"`
+	Name            string `json:"name,omitempty"`
+}
+
+// TeamsAccount returns teams account information with internal and external id
+func (api *API) TeamsAccount(ctx context.Context, accountID string) (TeamsAccount, error) {
+	return api.teamsAccount(ctx, accountID)
+}
+
+// TeamsAccountConfiguration returns teams account configuration
+func (api *API) TeamsAccountConfiguration(ctx context.Context, accountID string) (TeamsConfiguration, error) {
+	return api.teamsAccountConfiguration(ctx, accountID)
+}
+
+// TeamsAccountUpsertConfiguration upserts teams account configuration
+func (api *API) TeamsAccountUpsertConfiguration(ctx context.Context, accountID string, config TeamsConfiguration) (TeamsConfiguration, error) {
```

Comment:
`s/Upsert/Update`? the term upsert definitely leaves some ambiguity as to what is happening internally. if we only offer `HttpPatch`, perhaps Update is more appropriate here.